pkg/appStore/installedApp/service/AppStoreDeploymentService.go

```diff
@@ -851,8 +825,15 @@ func (impl *AppStoreDeploymentServiceImpl) UpdateInstalledApp(ctx context.Contex
 			return nil, err
 		}
 	}
+	err1 := impl.UpdatePreviousDeploymentStatusForAppStore(upgradeAppRequest, triggeredAt, err)
+	if err1 != nil {
+		impl.logger.Errorw("error while update previous installed app version history", "err", err, "installAppVersionRequest", upgradeAppRequest)
+		//if installed app is updated and error is in updating previous deployment status, then don't block user, just show error.
+	}
 
-	return upgradeAppRequest, nil
+	err = impl.attributesService.UpdateKeyValueByOne(bean2.HELM_APP_UPDATE_COUNTER)
```

Comment:
this value should be update irrespective of update is success/failure, move this back to rest handler